### PR TITLE
feat(R-105372): Implemented Basic File Structure

### PR DESCRIPTION
**Pull request by shubhamadhk15**
=>Basic Backbone File Structure 
=>Basic Functionalities-CRUD
=>Database Connection with PostgreSQL

### CODE REVIEW

File: .classpath
Comment (sutirthak) [inline .classpath:1]: Remove this file

File: .project
Comment (sutirthak) [inline .project:1]: Remove this file

File: README.md
Comment (sutirthak) [inline README.md:1]: Why did you remove this file?

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>3.2.2</version>
+		<relativePath/> <!-- lookup parent from repository -->
+	</parent>
+	<groupId>com.projects</groupId>
+	<artifactId>Employee_management_system</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
```
Comment (sutirthak) [inline pom.xml]: Use version 0.1.0

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>3.2.2</version>
+		<relativePath/> <!-- lookup parent from repository -->
+	</parent>
+	<groupId>com.projects</groupId>
+	<artifactId>Employee_management_system</artifactId>
```
Comment (sutirthak) [inline pom.xml]: Do not use underscore for artifact id

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	<version>0.0.1-SNAPSHOT</version>
+	<name>Employee_management_system</name>
+	<description>Sring Project for Employee Management System</description>
```
Comment (sutirthak) [inline pom.xml]: Correct spelling Sring -> Spring

File: src/main/java/com/qinvent/hr/EmployeeManagementSystemApplication.java
Comment (sutirthak) [inline src/main/java/com/qinvent/hr/EmployeeManagementSystemApplication.java:1]: Rename this to Application.

File: src/main/java/com/qinvent/hr/models/Employee.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	public Employee(int id, String name, int age, String type, float salary, String location) {
+		super();
+		this.id = id;
+		this.name = name;
+		this.age = age;
+		this.type = type;
+		this.salary = salary;
+		this.location = location;
+	}
+
```
Comment (sutirthak) [inline src/main/java/com/qinvent/hr/models/Employee.java]: Remove blank spaces

File: src/main/java/com/qinvent/hr/repository/EmployeeRepo.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package com.qinvent.hr.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
```
Comment (sutirthak) [inline src/main/java/com/qinvent/hr/repository/EmployeeRepo.java:3]: Remove the file if not in use

File: src/main/java/com/qinvent/hr/service/EmployeeServ.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
+package com.qinvent.hr.service;
+
+import java.util.List;
+import java.util.Optional;
+
+import com.qinvent.hr.models.Employee;
+
+public interface EmployeeServ {
+
```
Comment (sutirthak) [inline src/main/java/com/qinvent/hr/service/EmployeeServ.java]: Remove blank spaces

File: src/main/java/com/qinvent/hr/serviceIMPL/EmployeeServIMPL.java
Comment (sutirthak) [inline src/main/java/com/qinvent/hr/serviceIMPL/EmployeeServIMPL.java:1]: Move impl file under the same package Service

File: src/main/resources/application.properties [hunk 7]
```diff
@@ -0,0 +1,10 @@
+
+server.port=8888
+
+spring.datasource.url=jdbc:postgresql://localhost:5432/Employee
+spring.datasource.username=postgres
+spring.datasource.password=post12345
```
Comment (sutirthak) [inline src/main/resources/application.properties:6]: Properties file should never store the password

File: target/classes/application.properties
Comment (sutirthak) [inline target/classes/application.properties:1]: Remove target folder and add target folder in .gitignore for future mistakes

File: pom.xml [hunk 8]
```diff
@@ -9,10 +9,10 @@
 		<relativePath/> <!-- lookup parent from repository -->
 	</parent>
 	<groupId>com.projects</groupId>
-	<artifactId>Employee_management_system</artifactId>
-	<version>0.0.1-SNAPSHOT</version>
+	<artifactId>EmployeeManagementSystem</artifactId>
+	<version>0.1.0-SNAPSHOT</version>
```
Comment (sutirthak) [inline pom.xml]: Do not use SNAPSHOT at the end. Just 0.1.0

File: .classpath
Comment (shubhamadhk15) [reply .classpath:1]: Removed

File: .project
Comment (shubhamadhk15) [reply .project:1]: Removed

File: README.md
Comment (shubhamadhk15) [reply README.md:1]: I guess it was somehow deleted by mistake. Added the file to the branch

File: pom.xml (hunk 1 above)
Comment (shubhamadhk15) [reply pom.xml]: Updated

File: pom.xml (hunk 2 above)
Comment (shubhamadhk15) [reply pom.xml]: Updated artifact Id as : EmployeeManagementSystem

File: pom.xml (hunk 3 above)
Comment (shubhamadhk15) [reply pom.xml]: Updated

File: src/main/java/com/qinvent/hr/EmployeeManagementSystemApplication.java
Comment (shubhamadhk15) [reply src/main/java/com/qinvent/hr/EmployeeManagementSystemApplication.java:1]: Renamed to Application.java

File: src/main/java/com/qinvent/hr/models/Employee.java (hunk 4 above)
Comment (shubhamadhk15) [reply src/main/java/com/qinvent/hr/models/Employee.java]: Updated

File: src/main/java/com/qinvent/hr/service/EmployeeServ.java (hunk 6 above)
Comment (shubhamadhk15) [reply src/main/java/com/qinvent/hr/service/EmployeeServ.java]: Updated

File: src/main/java/com/qinvent/hr/serviceIMPL/EmployeeServIMPL.java
Comment (shubhamadhk15) [reply src/main/java/com/qinvent/hr/serviceIMPL/EmployeeServIMPL.java:1]: Moved the serviceIMPL class to the same service package

File: target/classes/application.properties
Comment (shubhamadhk15) [reply target/classes/application.properties:1]: done

File: pom.xml (hunk 8 above)
Comment (shubhamadhk15) [reply pom.xml]: updated

File: README.md [hunk 9]
```diff
@@ -1 +1,29 @@
-This repository is used to keep hr-activities code
+## Scope
+The Employee Management System is designed to manage the employee data 
+with an organisation. t allows users to perform various CRUD (Create, Read, Update, Delete) operations
```
Comment (sutirthak) [inline README.md]: Typo here t => It

File: README.md [hunk 10]
```diff
@@ -1 +1,29 @@
...
+* Java Development Kit (JDK)
+* IntelliJ Idea or Spring Tools Suite
+* Post GreSQL
+
+## Building the Application :
+* Clone into the GitHub Repostitory : git clone https://github.com/sutirthak/hr-activities.git
+* Open the Maven Project in the IDE
+* Create a Employee Database in PostGreSQL
+* Update the application.properties file with Database Credentials
+* Run Application.java
```
Comment (sutirthak) [inline README.md]: Maven does not depend on IDE. 
So give the command how it can be compiled and run like:
mvn clean install => to build the jar file
Then java -jar <executable jar file name> to run the jar

File: README.md (hunk 9 above)
Comment (shubhamadhk15) [reply README.md]: Fixed

File: README.md (hunk 10 above)
Comment (shubhamadhk15) [reply README.md]: Updated the documentation and running sucessfully.

**Comment by shubhamadhk15** [comment]
Created the Javadoc and saved in hr-activities/doc folder

**Review by sutirthak — APPROVED** [review]
